model-serving/content.md

```diff
@@ -0,0 +1,292 @@
+input_data = iris["data"]
+target_data = iris["target"]
+model = DecisionTreeClassifier()
+model.fit(input_data, target_data)
+
+# Note: in the following line, insert the project token shown on your dashboard page.
+mflux_ai.init("your_project_token_goes_here")
+mlflow.sklearn.log_model(model, "model")
+
+```
+
+Now, run this file: ```python train_model.py```.
+
+### REST API with Flask
+
+Create a file app.py inside a folder named ```app``` and paste the following code:
+
+```python
+import mflux_ai
+import mlflow.sklearn
+import numpy as np
+import pickle
+from flask import Flask, request, send_from_directory
+from flask import jsonify
+
+app = Flask(__name__, static_folder="app")
+
+# Note: in the following line, insert the project token shown on your dashboard page.
+mflux_ai.init("your_project_token_goes_here")
+
+model = mlflow.sklearn.load_model(
+    "s3://mlflow/0/RUN_ID_GOES_HERE/artifacts/model"
+)
+
+
+@app.route("/predict/", methods=['POST'])
+def predict():
+    response = {"success": False}
+    if request.method == "POST":
+        if request.data:
+            data = request.json.get('data')
+            predictions = model.predict(np.asarray(data))
+            response["success"] = True
+            response["predictions"] = predictions.tolist()
+
+    return jsonify(response)
+
+```
+
+In ```app.py```, replace ```RUN_ID_GOES_HERE``` with the actual run id that you found in the model tracking UI.
+
+Let's go through the code. The first code snippet imports the packages and initalizes the Flask application
+
+
+```python
+import mflux_ai
+import mlflow.sklearn
+import numpy as np
+import pickle
+from flask import Flask, request, send_from_directory
+from flask import jsonify
+
+app = Flask(__name__, static_folder="app")
+```
+
+
+Next, we load our trained model from MFlux.ai.
+```python
+# Note: in the following line, insert the project token shown on your dashboard page.
+mflux_ai.init("your_project_token_goes_here")
+
+model = mlflow.sklearn.load_model(
+    "s3://mlflow/0/RUN_ID_GOES_HERE/artifacts/model"
+)
+```
+
+
+We define the ```predict```function which processes any requests to
+the ```/predict/``` endpoint.
+```python
+@app.route("/predict/", methods=['POST'])
+def predict():
+    response = {"success": False}
+    if request.method == "POST":
+        if request.data:
+            data = request.json.get('data')
+            predictions = model.predict(np.asarray(data))
+            response["success"] = True
+            response["predictions"] = predictions.tolist()
+
+    return jsonify(response)
+```
+The function takes the incoming data and feeds it into the model. It
+then returns the predictions to the client in JSON format.
+
+
+#### Dockerfile
+
+Make a ```Dockerfile``` and paste the following code:
+```Dockerfile
+FROM continuumio/miniconda3:4.6.14
+
+RUN pip install pyuwsgi==2.0.18
+
+RUN mkdir /app/
+WORKDIR /app
+
+COPY ./requirements.txt /app/requirements.txt
+RUN pip install -r requirements.txt
```

Comment:
We could also add the `--no-cache-dir` option here, to reduce the final size of the docker image